cogs/heartbeat/heartbeat.py

```diff
@@ -48,14 +48,20 @@ async def _loop(self):
                                 "Something went wrong, we got HTTP code %s",
                                 resp.status,
                             )
-            except asyncio.CancelledError as e:
-                LOGGER.error(
-                    "The background task got cancelled! If the cog was reloaded, "
-                    "this can be safely ignored",
-                    exc_info=True,
-                )
-            except:
-                LOGGER.error("Something went horribly wrong!", exc_info=True)
+            except BaseException as e:
+                if isinstance(e, asyncio.CancelledError):
+                    LOGGER.error(
+                        "The background task got cancelled! If the cog was reloaded, "
+                        "this can be safely ignored",
+                        exc_info=True,
+                    )
+                else:
+                    LOGGER.error("Something went horribly wrong!", exc_info=True)
+                if not session.closed:
+                    await session.close()
+                break
+        if not session.closed:
+            await session.close()
```

Comment:
A few thoughts here:
1. If we're going with break here, we could get rid of the first `if not session.closed` since you'll hit the same code block right after you exit exit anyways.
2. That being said, this infinite loop does a simple GET request, so in the event that it fails for some reason (e.g. it times out, connection between bot and Internet is interrupted, etc), we immediately exit this loop and will never call the function again until we reload the cog, even if the anomaly resolves itself soon after. I think we should keep trying to punch the heartbeat even if we run into errors.

Reply:
Switched this PR's status to **draft**.

1. First point valid. We can get rid of that.
2. I did not consider any recovery logic and so we certainly could write some more just so the task doesn't end until being  cancelled. Perhaps we can just plainly get rid of `break`, but would it spam logging in case the loop fails multiple times in a certain window of time?